demo/src/Components/BottomTabBar.tsx

```diff
@@ -28,9 +28,12 @@ const BottomTabBar = (props: HvComponentProps) => {
       );
       return;
     }
-    ctx.setElementProps?.(navigator, props);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [navigator, props]); // Exclude ctx from dependencies to avoid infinite loop
+    // To avoid rendering loops, we only set the element props once
+    if (props.element.getAttribute('registered') !== 'true') {
```

Comment:
What happens when the screen triggers a `reload` action and potentially gets a new bottom tab bar from the server? Is that handled correctly?

Reply:
Yes, it is handled correctly - the video I attached to the PR description showcases this.